public/Test-AdfLinkedService.ps1

```diff
@@ -9,27 +9,37 @@ function Test-AdfLinkedService {
         [String] $ResourceGroupName,
         [parameter(Mandatory = $true)] 
         [String] $SubscriptionID,
-        [parameter(Mandatory = $true)] 
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
         [String] $TenantID,
-        [parameter(Mandatory = $true)] 
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
         [String] $ClientID,
-        [parameter(Mandatory = $true)] 
-        [String] $ClientSecret
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
+        [String] $ClientSecret,
+        [parameter(Mandatory = $true, ParameterSetName="AzRestMethod")]
+        [switch] $preferAzRestMethod
```

Comment:
I can't see the role of this switch

Reply:
It is used https://github.com/SQLPlayer/azure.datafactory.tools/pull/151/files#diff-ad136628efb917d1afaf8ff1157f12bd127c9cbbd4fed91fce0fa56a8885c83bR22 and here https://github.com/SQLPlayer/azure.datafactory.tools/pull/151/files#diff-ad136628efb917d1afaf8ff1157f12bd127c9cbbd4fed91fce0fa56a8885c83bR32